src/dmd/dtoh.d

```diff
@@ -1548,7 +1541,7 @@ public:
             printf("[AST.NullExp enter] %s\n", e.toChars());
             scope(exit) printf("[AST.NullExp exit] %s\n", e.toChars());
         }
-        buf.writestring("_d_null");
+        buf.writestring("nullptr");
```

Comment:
`nullptr` is a C++11 feature. Do we still need to support C++98?

Reply:
I comment on this in the heading ;)

We removed the code to support pre-C++11 `wchar`/`dchar` types mid last year, so our `extern(C++)` currently has an affinity for C++11.

In the event the user tries to build with pre-C++11 compiler, they may simply `#define nullptr NULL` before the include, and we'll be leaning further into this pattern to `#define` configuration details before including the header soon... `enum`s will be a bunch of macros which users can control.

Reply:
Perhaps it would be best to check the value of `params.cplusplus` and choose accordingly?

Reply:
I thought this initially; but I'm not sure we should do that...
When we build with `-extern-std`, we're saying ABI stuff about what we link to within our project. It specifies details about the C++ code we intend to link against.

But the header file that's emit is not necessarily meant to be used that way, I think it's more likely that the header is to be used by C++ projects that intend to link against our D code... so it's the reverse relationship, and I'm not sure the same compile option should be used to specify that.
I'm more inclined to emit a standard output, and use macros where C++ has evolved (like enums), allowing the client app to define those details before including the header.

I could be convinced otherwise though.

Reply:
I mean, to be clear, I'm not sure choosing to compile with a different `-extern-std` should allow the C++ client code to become broken and require changes because the header changed.

Reply:
C++ code we link against and c++ interfaces we provide already have to adhere to the same standard, though.  Consider the case where you pass an `std::vector` to a c++ library; if the vector type is defined in d code as the c++03 type, but the library is c++11, there will be a mismatch.

Reply:
Here is one c++ project that uses `-std=gnu++98` for bootstrapping. :wave: